java-frontend/src/main/java/org/sonar/java/cfg/CFG.java

```diff
@@ -840,6 +855,28 @@ private void buildNewClass(NewClassTree tree) {
     build(Lists.reverse(tree.arguments()));
   }
 
+  private void handleExceptionalPaths(Symbol symbol) {
+    TryStatement tryStatement = enclosingTry.peek();
+    if(tryStatement != outerTry) {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png) Change this comparison to use the equals method. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](https://dory.sonarsource.com/sonarqube/coding_rules#rule_key=squid%3AS1698)